src/components/SystemMessageList/SystemMessageList.tsx

```diff
@@ -0,0 +1,51 @@
+import React, {useEffect, useState} from 'react';
+import {
+  SystemMessageNotification,
+  SystemMessageNotificationTypes
+} from '../SystemMessageNotification/SystemMessageNotification';
+import {getSystemMessages} from '../../api';
+import {useMount} from '../../hooks/useMount';
+
+
+export interface SystemMessageList {
+  id: number;
+  type: SystemMessageNotificationTypes;
+  description: string;
+
+}
+
+const SystemMessagesList = () => {
+
+  const [systemMessageList, setSystemMessageList] = useState<SystemMessageList[]>([]);
+
+  const [dismissedMessage, setDismissedMessage] = useState<string[]>([]);
+
+  useEffect(() => {
+    window.sessionStorage.setItem('dismissedMessage', dismissedMessage.join());
+  }, ['dismissedMessage']);
+
+  useMount(async () => {
+    const messages = await getSystemMessages();
+    setSystemMessageList(messages);
+    const verifySessionID = window.sessionStorage.getItem('dismissedMessage').split(',');
+    if (verifySessionID) {
+      setDismissedMessage(verifySessionID);
+    }
+  });
+
+  return (
+    <div>
+      {systemMessageList.map((message: SystemMessageList) => {
+        if (!dismissedMessage.includes(message.id.toString())) {
```

Comment:
@AStiner I see your problem. You can't use an if statement in a return expression like that. React is expecting jsx (basically just HTML with javascript objects in it) so you have to use a ternary operator, like I suggest.

Reply:
Yep, figured it out thanks. Working on other error now with the windows session storage. TS2531: Object is possibly 'null'.

Reply:
I think if you put a ! after `dismissedMessage` it might fix that:

`!dismissedMessage!.includes(...`

Reply:
If not you'll probably have to extract this functionality into its own component